### PR TITLE
Add check itemframe when breaking

### PR DESCRIPTION
**Pull request by DoctaEnkoda**

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java [hunk 1]
```diff
@@ -66,4 +68,21 @@ public void monitorBlockBreak(BlockBreakEvent event) {
...
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+    public void onHangingBreakByEntityEvent(HangingBreakByEntityEvent event) {
+        if (event.getRemover() instanceof Player) {
+            if (event.getEntity() instanceof ItemFrame) {
+                Player player = (Player) event.getRemover();
+                ItemFrame itemFrame = (ItemFrame) event.getEntity();
+                User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+                Optional<Island> island = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(itemFrame.getLocation());
+                if (!island.isPresent()) return;
+                if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, PermissionType.BLOCK_BREAK)) {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java:80]: Use ifPresent method its better

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java [hunk 2]
```diff
@@ -66,4 +68,21 @@ public void monitorBlockBreak(BlockBreakEvent event) {
         });
     }
 
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+    public void onHangingBreakByEntityEvent(HangingBreakByEntityEvent event) {
+        if (event.getRemover() instanceof Player) {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java:73]: users can throw egg-snowball to itemframes so u also need check this for projectiles

**Review by MertUnverdi — APPROVED** [review]

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java (hunk 1 above)
Comment (DoctaEnkoda) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java:80]: To use it, you have to pass the plugin on Java 11 minimum, I also prefer it but cannot currently

Comment (DoctaEnkoda) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java:80]: ah but its ifPresent

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java:80]: nope its already exist on java 8 tho ?

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java:80]: also we already using it on everywhere

Comment (DoctaEnkoda) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java:80]: Ah yes, he is on Java 8 my bad,I will modify it tomorrow

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java (hunk 2 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockBreakListener.java:73]: Even arrows, if I remember correctly

**Review by dlsf — CHANGES_REQUESTED** [review]
See Slash's comments, this is not ready yet. 

Bonus points for applying the early return pattern and fixing the indentation.

**Review by MertUnverdi — APPROVED** [review]

**Pull request merged**